Reduce config's default for max TIs per query & enforce <= parallelism (#32572)
mistercrunch/Airflow: docs/apache-airflow/administration-and-deployment/scheduler.rst

  The scheduler will list and sort the DAG files to decide the parsing order.

- :ref:`config:scheduler__max_tis_per_query`
  The batch size of queries in the scheduling main loop. This should not be greater than

>   ``core.parallelism``.If this is too high, SQL query performance may be impacted by
>   complexity of query predicate, and/or excessive locking.

Reviewer: ```suggestion
  ``core.parallelism``. If this is too high then SQL query performance may be impacted by
  complexity of query predicate and/or excessive locking.
```

  Additionally, you may hit the maximum allowable query length for your db.

>   Setting this to 0 for using the value of ``core.parallelism``.

Reviewer: ```suggestion
  Setting this to 0 will use the value of ``core.parallelism``.
```
